src/rebar3_hex.erl

```diff
@@ -29,6 +29,12 @@ task_args(State) ->
     {[{task, Task} | Args], _} = rebar_state:command_parsed_args(State),
     {Task, Args}.
 
+sub_command(State) ->
```

Comment:
I'm not terribly sure this is the best way to skin this cat, I don't see harm in it either. As a counter example, in rebar3_hex_repo we make use of `{Args, _} = rebar_state:command_parsed_args(State)` and `proplists:get_value(subcmd, Args, undefined)` but pretty sure they both do the same thing, maybe there's added checks and balances from rebar3 if we get the subcmd from the `subcmd` tuple though.

Reply:
Ignore the above, `subcmd` is not reliably. However, I'm thinking about doing away with "lose arguments". When we use switches we can get all the arguments in a normalize form, lose arguments (such as the key name argument to the revoke sub command) are well, lose. We end up having to have special cases for these and we rely that they are specified in exactly the right order. 

So, instead of `rebar3 hex key revoke bar` we would just switch to `rebar3 hex key revoke --key-name bar`. The latter is less ergonomic but, I think the trade off is worth it.

Reply:
I'm going to push this change up and we can decide to keep it or not.